src/message.lisp

```diff
@@ -133,6 +306,456 @@
              `(:reply_to_message_id ,reply-to-message-id)))))
     (make-request bot "sendMessage" options)))
 
+(defmethod send-photo (bot chat (photo string)
+                       &key caption parse-mode caption-entities
+                         disable-notification protect-content reply-to-message-id
+                         allow-sending-without-reply reply-markup)
+  "A method for photo sending based on photo ID.
+
+The file-based method does not work yet.
+
+https://core.telegram.org/bots/api#sendphoto"
+  (log:debug "Sending photo" chat photo)
+  (let ((options
+          (append
+           `(:|chat_id| ,(get-chat-id chat)
+              :|photo| ,photo)
+           (when caption
+             `(:|caption| ,caption))
+           (when caption-entities
+             `(:|caption_entities| ,caption-entities))
+           (when parse-mode
+             `(:|parse_mode| ,parse-mode))
+           (when disable-notification
+             `(:disable_notification ,disable-notification))
+           (when protect-content
+             `(:|protect_content| ,protect-content))
+           (when reply-to-message-id
+             `(:reply_to_message_id ,reply-to-message-id))
+           (when allow-sending-without-reply
+             `(:|allow_sending_without_reply| ,allow-sending-without-reply))
+           (when reply-markup
+             `(:|reply_markup| ,reply-markup)))))
```

Comment:
Here and at other places of the pull, it is better to use `(list ...)` instead of quotes, for the reason I've described in the previous pull: https://github.com/40ants/cl-telegram-bot/pull/9#discussion_r895079787

I see you are constructing such options list in many places. Probably the code can be made simpler, if we'll modify `make-request` signature like this:

```lisp
(defun make-request (bot name &rest options &key (streamp nil) (timeout 3) &allow-other-keys)
```

And do:

```list
(loop for (key value) on options by 'cddr
      when value
        collect (kebab:to-snake-case key)
        and
          collect value)
```

inside.

Of cause you'll have to `(alexandria:remove-from-plistf options :timeout :streamp)`

After that you can get rid of these multiple when checks and will just call make-request with proper arguments:

```lisp
(make-request bot "sendPhoto"
              :chat-id (get-chat-id chat)
              :photo photo
              :caption caption
              :caption-entities caption-entities
              ...)
```